docs/dvc.md

```diff
@@ -0,0 +1,62 @@
+# Data Version Control
+
+Machine Learning projects require both code and data to build a model. Both the code and the data needs to be version controlled to ensure that results are reproducible at any point in time. However, data presents two version control challenges:
+
+1. Large datasets - often the training dataset is too large to fit in a Git repository.
+2. Slow pipeline stages - Training a model on a large dataset can take a significant amount of time. It's painful to be forced to retrain a model if the training data, code, and outputs are up-to-date already.
+
+Data Version Control ([DVC](https://dvc.org/)) is a Python package that solves the above ML-specific project challenges and is installed as a dependency within whisk projects. It's a lightweight, flexible tool that aligns closely with git commands.
+
+## Enabling DVC
+
+Type [`whisk dvc setup`](cli_reference.html#whisk-dvc-setup) to enable DVC within your project. After setup completes, you need a place to store your data. whisk includes a set of commands to automate the setup of an S3 [DVC remote](https://dvc.org/doc/command-reference/remote). To create a remote:
+
+```
+whisk s3 create
+whisk dvc remote-add
+```
+
+This creates an Amazon S3 bucket named `whisk-<project_name>` and creates a DVC remote that uses the S3 bucket.
+
+## Using DVC
+
+The DVC website has [excellent documentation](https://dvc.org/doc) on using DVC. Please refer to their website.
+
+## Example Project
+
+The [Real or Not? NLP with Disaster Tweets](https://github.com/whisk-ml/disaster_tweets) whisk project uses DVC to version control the data download and training stages. This keeps the large dataset and model artifacts outside of version control. As the training stage takes ~20 minutes on a laptop, this can save a significant amount of time when bootstrapping the project. New users can immediately begin using the model after running `dvc pull` versus waiting for the training to complete.
+
+## Heroku deployment
+
+If your app stores artifacts in DVC, those files will need be available by default when the app is [deployed to Heroku](heroku.html). To ensure the artifacts are available, we suggest adding a `.profile` file to the root of your project to pull these artifacts.
+
+[Real or Not? NLP with Disaster Tweets](https://github.com/whisk-ml/disaster_tweets/blob/master/.profile) uses the following `.profile` to do this:
+
+    # .profile
+
+    git init # Needed as dvc pull will not work if this isn't a Git repo.
+    dvc pull train.dvc # Grab only the training output, not all files (don't need raw data)
+
+Heroku [runs the .profile](https://devcenter.heroku.com/articles/dynos#the-profile-file) file before executing the dyno’s command. This ensure that every dyno type has access to the needed model artifacts.
```

Comment:
@itsderek23 - curious... this loads when the dyno starts up, but not during building? So this won't influence the slug limit, but will run up against dyno memory limits? I wasn't able to decipher that from the heroku docs, but maybe I was looking at the wrong place

Reply:
> this loads when the dyno starts up, but not during building? 

Correct.

> So this won't influence the slug limit, but will run up against dyno memory limits?

I don't think it will impact dyno memory limits. The data isn't loaded into memory. It's saved to disk. There are GBs available, which should be fine for most model artifacts: https://help.heroku.com/DGUDV63H/deprecated-how-much-disk-space-on-the-dyno-can-i-use.

Reply:
Ah nice sounds good. I guess it will only impact memory limits once the model or artifacts are loaded in (say in the `__init__` within `model.py`)? Mostly trying to understand when models will hit size limits on Heroku (outside of optimizing the slug or loading data to disk).

Reply:
Good question. I think you wiill see memory increase when the models are loaded from disk (ex: via pickle) within the Flask app. I don't know how tightly the increase is aligned to the disk size (imagine it is in the same ballpark).